hooks/conan-center.py

```diff
@@ -623,6 +624,12 @@ def test(out):
                     out.error("The file '{}' ends with a dot. Please, remove the dot from the end."
                               .format(file, disallowed_chars))
 
+    @run_test("KB-H059", output)
+    def test(out):
+        class_name = type(conanfile).__name__
+        if class_name in ("LibnameConan", "ConanFileDefault"):
+            out.error("Class name '{}' is not allowed. For example, use '{}Conan' instead.".format(class_name, conanfile.name))
```

Comment:
Your suggestion might be invalid: e.g. `android-ndkConan`.
How about the following?
```suggestion
            camel_name = "".join(s.title() for s in re.split("[^a-zA-Z]", conanfile.name))
            out.error("Class name '{}' is not allowed. For example, use '{}Conan' instead.".format(class_name, camel_name))
```

(make sure `re` is imported)